pandas/core/generic.py

```diff
@@ -10416,44 +10438,54 @@ def _doc_parms(cls):
           Aug      113
 dtype: int64
 
->>> s.sum()
-478
+>>> s.%(stat_func)s()
+%(default_output)s
 
-Sum using level names, as well as indices
+%(verb)s using level names, as well as indices.
 
->>> s.sum(level='city')
+>>> s.%(stat_func)s(level='city')
 city
-London      136
-New York    342
+London      %(named_level_output_0)s
+New York    %(named_level_output_1)s
 dtype: int64
 
->>> s.sum(level=1)
+>>> s.%(stat_func)s(level=1)
 month
-Jun    159
-Jul    152
-Aug    167
+Jun    %(indexed_level_output_0)s
+Jul    %(indexed_level_output_1)s
+Aug    %(indexed_level_output_2)s
 dtype: int64
-
-By default, the sum of an empty or all-NA Series is ``0``.
-
->>> pd.Series([]).sum()  # min_count=0 is the default
-0.0
-
-This can be controlled with the ``min_count`` parameter. For example, if
-you'd like the sum of an empty series to be NaN, pass ``min_count=1``.
-
->>> pd.Series([]).sum(min_count=1)
-nan
-
-Thanks to the ``skipna`` parameter, ``min_count`` handles all-NA and
-empty series identically.
-
->>> pd.Series([np.nan]).sum()
-0.0
-
->>> pd.Series([np.nan]).sum(min_count=1)
-nan
-"""
+%(additional)s"""
+
+_shared_docs['sum_multi_index'] = dict(stat_func='sum', verb='Sum',
+                                       default_output='478',
+                                       named_level_output_0='136',
+                                       named_level_output_1='342',
+                                       indexed_level_output_0='159',
+                                       indexed_level_output_1='152',
+                                       indexed_level_output_2='167',
+                                       additional=_sum_na)
+
+_shared_docs['max_multi_index'] = dict(stat_func='max', verb='Maximize',
+                                       default_output='117',
+                                       named_level_output_0=' 54',
+                                       named_level_output_1='117',
+                                       indexed_level_output_0='112',
+                                       indexed_level_output_1='117',
+                                       indexed_level_output_2='113',
+                                       additional='')
+
+_shared_docs['min_multi_index'] = dict(stat_func='min', verb='Minimize',
+                                       default_output='35',
+                                       named_level_output_0=' 35',
+                                       named_level_output_1='112',
+                                       indexed_level_output_0='47',
+                                       indexed_level_output_1='35',
+                                       indexed_level_output_2='54',
+                                       additional='')
+
+_sum_examples = _shared_docs['stat_func_example'] % \
+    _shared_docs['sum_multi_index']
```

Comment:
I think it's better if we do:
```
_sum_examples = _shared_docs['stat_func_example'].format(
    stat_func='sum', verb='Sum', default_ouput='478',
    ...)
```

That will makes things simpler, having some `_shared_docs` with strings, and some with dicts, was quite confusing to me, took a decent amount of time to understand.

I think it'll probably make things simpler to use `'...{stat_func}...'.format(stat_func='sum')` than the one with a `%` in this case (but if it doesn't make things simpler, you can leave the old one.

I'd prefer an example where it's faster to understand the results. For example, if I show you `op(2, 3) == 5` it's quite easy to see that the `op` is summing. But with higher magnitudes, like the `op(112, 47)` it takes more time to make the math and make sure you're understanding the examples correctly.

We've been using in many places examples with animals and their number of legs, and things like that. I think it's probably a good idea to use them here.

For the `additional` variable, as it comes after the examples, I'd simply add them at the end like:
```
_sum_examples = _shared_docs['stat_func_example'].format(
    ....)
_sum_examples += '''
the rest of the example
'''
```
I think this makes things much easier to read and find too.

Reply:
.format() with keyword arguments is not allowed by old-style formatting. I propose the following as the closest approximation to what you want:
```
_sum_examples = _shared_docs['stat_func_example'] % dict(
    stat_func='sum',
    verb='Sum',
    ...)
```